player.c

```diff
@@ -354,7 +355,16 @@ static void update_rg_scale(void)
 	if (!player_info_priv.ti || !replaygain)
 		return;
 
-	if (replaygain == RG_TRACK || replaygain == RG_TRACK_PREFERRED) {
+	if (replaygain == RG_SMART) {
+		if (play_library && aaa_mode == AAA_MODE_ALBUM 
+			&& !shuffle && !cmus_queue_active()) {
+			gain = player_info_priv.ti->rg_album_gain;
+			peak = player_info_priv.ti->rg_album_peak;
+		} else {
+			gain = player_info_priv.ti->rg_track_gain;
+			peak = player_info_priv.ti->rg_track_peak;
+		}
+	} else if (replaygain == RG_TRACK || replaygain == RG_TRACK_PREFERRED) {
```

Comment:
It would be neater to extend this condition instead.
```
if (RG_TRACK or RG_TRACK_PREFERRED or avoid_album_gain)
```
where avoid album would be set at the start of the function as
```
aviod_album_gain = RG_SMART and (shuffle or queue_active)
```

Reply:
Looks good, but we also need to check if we're playing from a playlist instead of the library.